### DISCUSSION TITLE
No plate recognized

### DISCUSSION THREAD

**Discussion by ablyes** [blakeblackshear/frigate#7714] [General]
Hi all, 
I configured Frigate to recognize persopn, car, plate. 
Plate is not working. I have 0 plate in the events. 
Should I add something to the config file ?
I already pushed some images to frigate+ for the plates. I don't know what should I expect, and if this may help me.
Thank you for your help. 
Lyes A

**Comment by NickM-27** [comment]
What model are you using? The default model does not support license_plate as an object and Frigate+ models are not out yet

**Reply by ablyes** [reply]
Hi NickM, 
I didn't set any particular model. 
I believe frigate is using the default embedded model ?

Q1/ Could you advise me a particular model for plates ?
Q2/ Question that has nothing to do with this topic :
 I have a google Coral (detected), but i still can see the container taking 200% of CPU (2 CPU of my synology 918+ wich is a QUAD-CORE)

```
mqtt:
  enabled: True
  host: 192.168.1.100
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: "{FRIGATE_MQTT_USER}"
  password: "{FRIGATE_MQTT_PASSWORD}"
  stats_interval: 60

detectors:
  coral:
    type: edgetpu
    device: usb

database:
  path: /media/frigate/frigate.db


ffmpeg:
  output_args:
    record: preset-record-generic-audio-aac


logger:
  default: info
  logs:
    frigate.event: debug


birdseye:
  mode: objects
  width: 1920
  height: 1080
  restream: true


detect:
  width: 640
  height: 360
  fps: 7
  max_disappeared: 18
  stationary:
    interval: 12



objects:
  track:
    - person
    - plate 
    - car



record:
  enabled: true
  retain:
    days: 7
    mode: motion
  events:
    pre_capture: 3
    post_capture: 3
    retain:
      default: 14
      mode: active_objects
      objects:
        person: 30


snapshots:
  enabled: true
  retain:
    default: 30


rtmp:
  enabled: False


go2rtc:
  rtsp:
    username: "admin"
    password: "XX"
  streams:
    sonnette: 
      - 'rtsp://admin:XX@192.168.1.112/cam/realmonitor?channel=1&subtype=0#backchannel=0'
      - 'ffmpeg:sonnette#audio=opus'
    portail:
      - rtsp://admin:XX@192.168.1.110/Streaming/channels/101
      - isapi://admin:XX@192.168.1.110:80/
    est:
      - rtsp://admin:XX@192.168.1.111/Streaming/channels/101


cameras:
  sonnette:
    enabled: True
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/sonnette
          roles:
            - detect
            #- record
            #- rtmp
    motion:
      mask:
        - 640,0,0,0,0,209,640,237
    zones:
      zone_plaque:
        coordinates: 0,360,424,360,421,214,0,200
        objects:
          - car
          - plate
    detect:
      width: 1920
      height: 1080
    best_image_timeout: 60
    ui:
      order: 0
      dashboard: True
  portail:
    enabled: True
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/portail
          roles:
            - detect
    best_image_timeout: 60
    ui:
      order: 1
      dashboard: True
  est:
    enabled: True
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/est
          roles:
            - detect
    motion:
      mask:
        - 640,153,640,0,496,0
        - 135,21,0,22,0,127
        - 640,360,586,360,640,293
        - 0,360,80,360,0,244
    best_image_timeout: 60
    ui:
      order: 2
      dashboard: True

ui:
  live_mode: webrtc
  use_experimental: False
  time_format: browser
  date_style: short
  time_style: medium

telemetry:
  version_check: True

**Reply by NickM-27** [reply]
Okay, the default model (which uses the COCO dataset) doesn't support license_plate

> Could you advise me a particular model for plates ?

I am not aware of a single model that has all labels you want

> Q2/ Question that has nothing to do with this topic :
I have a google Coral (detected), but i still can see the container taking 200% of CPU (2 CPU of my synology 918+ wich is a QUAD-CORE)

https://github.com/blakeblackshear/frigate/issues/3860

**Reply by ablyes** [reply]
Q1
I'll look for another model. 
The documentation doesn't say a word about the need of a specific model for plate :( 

Q2
ok, i reduced the resolution to 1080 x 720 for the detection. 
I'm not sure what to add for the hardware acceleration.

I know that the VAINFO button of frigate, is showing (advising) this:

vainfo: VA-API version: 1.17 (libva 2.10.0)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 23.1.1 ()
vainfo: Supported profile and entrypoints
VAProfileNone : VAEntrypointVideoProc
VAProfileNone : VAEntrypointStats
VAProfileMPEG2Simple : VAEntrypointVLD
VAProfileMPEG2Main : VAEntrypointVLD
VAProfileH264Main : VAEntrypointVLD
VAProfileH264Main : VAEntrypointEncSlice
VAProfileH264Main : VAEntrypointFEI
VAProfileH264Main : VAEntrypointEncSliceLP
VAProfileH264High : VAEntrypointVLD

**Reply by NickM-27** [reply]
Add `preset-vaapi`

**Reply by ablyes** [reply]
Seems to work ! Thank you. 
First try: 
CPU **200% -> 175%**
changes done : fps 7 to 6 fps, 1920x1080 to 1080x720

Second try:
CPU **175% -> 135%**
changes done : adding preset-vaapi

**Is it acceptable for a Synology 918+ with a google coral ?**

**Reply by NickM-27** [reply]
You might want to try the i965 driver instead

**Reply by ablyes** [reply]
I changes the configuration for this 
```
ffmpeg:
  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
    - -hwaccel_output_format
    - yuv420p
  output_args:
    record: preset-record-generic-audio-aac
```
I'm around 130% of cpu usage, but i think i can do better, because i still see this error:
` ERROR   : Unable to poll intel GPU stats: Failed to initialize PMU! (Invalid argument)`

Container options
<img width="1243" alt="image" src="https://github.com/blakeblackshear/frigate/assets/2130374/36843331-6c61-4c14-9d0b-69c039de37e8">

**Reply by NickM-27** [reply]
You should not be using manual args, preset-vaapi is preferred. Like I suggested, look at using the i965 driver as the docs suggest https://docs.frigate.video/configuration/hardware_acceleration#via-vaapi

> I'm around 130% of cpu usage, but i think i can do better, because i still see this error:
 ERROR   : Unable to poll intel GPU stats: Failed to initialize PMU! (Invalid argument)

This has nothing to do with performance, just stats for the GPU. It is not supported on synology

**Reply by ablyes** [reply]
ok, got the point. 
- I removed the DEVICES, just let LIBVA_DRIVER_NAME.
- no manual parameters. 
- the error is not an error in my case. 
Now the CPI is between 50 and 80% which is more than acceptable. 

When should we expect a new model for plates with Frigate+ ? 
What is the start point for me, if I want to build an easy one : person, car, plate. 
The funny fact, i can find some models for specific birds, but not plates.
